src/System.Text.Json/src/System/Text/Json/JsonParseObject.cs

```diff
@@ -0,0 +1,350 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Text.Utf8;
+
+namespace System.Text.Json
+{
+    public struct JsonParseObject
+    {
+        private byte[] _buffer;
+        private int _start;
+        private int _end;
+
+        private const int RowSize = 9;
+
+        public JsonParseObject(byte[] buffer, int start, int end)
+        {
+            _buffer = buffer;
+            _start = start;
+            _end = end;
+        }
+
+        public bool HasValue()
+        {
+            var typeCode = _buffer[_start + 8];
+
+            if (typeCode == (byte)JsonParser.JsonValueType.Object || typeCode == (byte)JsonParser.JsonValueType.Array)
+            {
+                int length = BitConverter.ToInt32(_buffer, _start + 4);
```

Comment:
We are already accessing buffer at index _start + 8 to grab the typecode. If a 'row' exists, it will definitely contain ROWSIZE bytes (9) or else the DB wasn't created properly.